tests/steps/given.ts

```diff
@@ -3,35 +3,75 @@ import { api } from '../shared/classes/api';
 import { Given, Then } from "@cucumber/cucumber";
 import GPS from '@/models/GPS';
 import ConnectMongoDB from '@/lib/mongodb';
+import AISShips from '@/models/AISShips';
 
-let apiResponseData;
-let databaseData;
+let apiResponseData_GPS;
+let databaseData_GPS;
+let apiResponseData_AISShips;
+let databaseData_AISShips;
 
 Given('I clear the database', async function () {
     const db = await ConnectMongoDB();
     await GPS.deleteMany();
 });
 
+Given('I clear the DB', async function () {
+    const db = await ConnectMongoDB();
+    await AISShips.deleteMany();
+});
+
 Given('I insert GPS data into the database', async function () {
     const gpsData = {
         latitude: 49.2243,
         longitude: 4.5552,
         speed: 30,
         heading: 45,
     };
-
     await GPS.create(gpsData);
 });
 
+Given('I insert AISShips data into the database', async function () {
+    const aisshipData = {ships: [
+        {
+            id: 1,
+            latitude: 12.3323,
+            longitude: 40.2313,
+            speed: 24,
+            heading: 42,
+        }
+    ]};
+    await AISShips.create(aisshipData);
+});
+
 Then('the response data matches the data in the database', async function () {
-    databaseData = await GPS.find({}).then(function(gps) {
+    databaseData_GPS = await GPS.find({}).then(function(gps) {
         let transformedGPS = gps.map((data) => data.toJSON())
         return transformedGPS;
     });
 
-    apiResponseData = api.response.data.data[0];
-    expect(apiResponseData.latitude).to.equal(databaseData[0].latitude, 'Data in the response does not match data in the database');
-    expect(apiResponseData.longitude).to.equal(databaseData[0].longitude, 'Data in the response does not match data in the database');
-    expect(apiResponseData.speed).to.equal(databaseData[0].speed, 'Data in the response does not match data in the database');
-    expect(apiResponseData.heading).to.equal(databaseData[0].heading, 'Data in the response does not match data in the database');
+    apiResponseData_GPS = api.response.data.data[0];
+
+    const propertiesToCompare = ['latitude', 'longitude', 'speed', 'heading'];
```

Comment:
Hint: You can dynamically determine the fields of an object using `Object.keys()` rather than manually typing each one :)   

https://www.w3schools.com/jsref/jsref_object_keys.asp